lib/pure/memfiles.nim

```diff
@@ -144,14 +170,16 @@ proc open*(filename: string, mode: FileMode = fmRead,
 
     # since the strings are always 'nil', we simply always call
     # CreateFileMappingW which should be slightly faster anyway:
-    result.mapHandle = createFileMappingW(
+    let mapHandle = createFileMappingW(
       result.fHandle, nil,
       if readonly: PAGE_READONLY else: PAGE_READWRITE,
       0, 0, nil)
 
-    if result.mapHandle == 0:
+    if mapHandle == nil:
```

Comment:
See https://msdn.microsoft.com/en-us/library/windows/desktop/aa366537(v=vs.85).aspx

```
Return value

If the function succeeds, the return value is a handle to the newly created file mapping object.
If the object exists before the function call, the function returns a handle to the existing object (with its current size, not the specified size), and GetLastError returns ERROR_ALREADY_EXISTS.
If the function fails, the return value is NULL. To get extended error information, call GetLastError.
```

`CreateFileMapping` function return NULL if fails, so check it with `== nil`. Since it may returns NULL, can't assign to `result.mapHandle` directly.